service/mapper/mapper.go

```diff
@@ -171,8 +170,17 @@ func (m *Mapper) Run() error {
 	// state commitment to the state commitment of the next finalized block. At
 	// that point, we can discard all other tries, as the path must continue
 	// from the state trie with the sealed state commitment.
+	once := &sync.Once{}
 Outer:
 	for {
+		// We want to check in this tight loop if we want to quit, just in case
+		// we get stuck on a timed out network connection.
+		select {
+		case <-m.stop:
+			break Outer
+		default:
+			// keep going
+		}
```

Comment:
I think the idiom [_tight loop_](https://en.wiktionary.org/wiki/tight_loop) is reserved for cases where we iterate really fast over few instructions.  Not sure if this qualifies :p But that's a mega-nitpick anyway